File: samples/AdventureWorksLT.WpfApp/BindingFactory.cs
using DevZest.Data.Presenters;
using DevZest.Data.Views;

namespace DevZest.Samples.AdventureWorksLT
{
    public static class BindingFactory
    {
        public static RowCompositeBinding<SalesOrderHeaderBox> BindToSalesOrderHeaderBox(this SalesOrderInfo _, bool isNew,
            out RowBinding<ForeignKeyBox> shipToAddressBinding, out RowBinding<ForeignKeyBox> billToAddressBinding)
        {
            var result = new RowCompositeBinding<SalesOrderHeaderBox>()
                .AddChild(_.FK_Customer.BindToForeignKeyBox(_.Customer, CustomerBox.RefreshAction), v => v._customer)
                .AddChild(shipToAddressBinding = _.FK_ShipToAddress.BindToForeignKeyBox(_.ShipToAddress, AddressBox.RefreshAction), v => v._shipTo)
                .AddChild(billToAddressBinding = _.FK_BillToAddress.BindToForeignKeyBox(_.BillToAddress, AddressBox.RefreshAction), v => v._billTo)
                .AddChild(_.OrderDate.BindToDatePicker(), v => v._orderDate)
                .AddChild(_.ShipDate.BindToDatePicker(), v => v._shipDate)
                .AddChild(_.DueDate.BindToDatePicker(), v => v._dueDate)
                .AddChild(_.PurchaseOrderNumber.BindToTextBox(), v => v._purchaseOrderNumber)
                .AddChild(_.AccountNumber.BindToTextBox(), v => v._accountNumber)
                .AddChild(_.ShipMethod.BindToTextBox(), v => v._shipMethod)
                .AddChild(_.CreditCardApprovalCode.BindToTextBox(), v => v._creditCardApprovalCode)
                .AddChild(_.Status.BindToComboBox(), v => v._status)
                .AddChild(_.OnlineOrderFlag.BindToCheckBox(), v => v._onlineOrderFlag)
                .AddChild(_.Comment.BindToTextBox(), v => v._comment);
            if (!isNew)
                result.AddChild(_.SalesOrderNumber.BindToTextBlock(), v => v._salesOrderNumber);
            return result;
        }

        public static RowCompositeBinding<SalesOrderFooterBox> BindToSalesOrderFooterBox<T>(this SalesOrderBase<T> _)
            where T : SalesOrderDetail, new()
        {
            return new RowCompositeBinding<SalesOrderFooterBox>()
                .AddChild(_.SubTotal.BindToTextBlock("{0:C}"), v => v._subTotal)
                .AddChild(_.Freight.BindToTextBox(), v => v._freight)
                .AddChild(_.TaxAmt.BindToTextBox(), v => v._taxAmt)
                .AddChild(_.TotalDue.BindToTextBlock("{0:C}"), v => v._totalDue);
        }

        public static RowCompositeBinding<AddressBox> BindToAddressBox(this Address _)
        {
            return new RowCompositeBinding<AddressBox>()
                .AddChild(_.AddressLine1.BindToTextBlock(), v => v._addressLine1)
                .AddChild(_.AddressLine2.BindToTextBlock(), v => v._addressLine2)
                .AddChild(_.City.BindToTextBlock(), v => v._city)
                .AddChild(_.StateProvince.BindToTextBlock(), v => v._stateProvince)
                .AddChild(_.CountryRegion.BindToTextBlock(), v => v._countryRegion)
                .AddChild(_.PostalCode.BindToTextBlock(), v => v._postalCode);
        }
    }
}

File: samples.vb/AdventureWorksLT.WpfApp/BindingFactory.vb
Imports DevZest.Data.Presenters
Imports DevZest.Data.Views

Public Module BindingFactory
    <System.Runtime.CompilerServices.Extension>
    Public Function BindToSalesOrderHeaderBox(e As SalesOrderInfo, isNew As Boolean, ByRef shipToAddressBinding As RowBinding(Of ForeignKeyBox),
                                              ByRef billToAddressBinding As RowBinding(Of ForeignKeyBox)) As RowCompositeBinding(Of SalesOrderHeaderBox)
        shipToAddressBinding = e.FK_ShipToAddress.BindToForeignKeyBox(e.ShipToAddress, AddressBox.RefreshAction)
        billToAddressBinding = e.FK_BillToAddress.BindToForeignKeyBox(e.BillToAddress, AddressBox.RefreshAction)
        Dim result = New RowCompositeBinding(Of SalesOrderHeaderBox)() _
            .AddChild(e.FK_Customer.BindToForeignKeyBox(e.Customer, CustomerBox.RefreshAction), Function(v) v._customer) _
            .AddChild(shipToAddressBinding, Function(v) v._shipTo) _
            .AddChild(billToAddressBinding, Function(v) v._billTo) _
            .AddChild(e.OrderDate.BindToDatePicker(), Function(v) v._orderDate) _
            .AddChild(e.ShipDate.BindToDatePicker(), Function(v) v._shipDate) _
            .AddChild(e.DueDate.BindToDatePicker(), Function(v) v._dueDate) _
            .AddChild(e.PurchaseOrderNumber.BindToTextBox(), Function(v) v._purchaseOrderNumber) _
            .AddChild(e.AccountNumber.BindToTextBox(), Function(v) v._accountNumber) _
            .AddChild(e.ShipMethod.BindToTextBox(), Function(v) v._shipMethod) _
            .AddChild(e.CreditCardApprovalCode.BindToTextBox(), Function(v) v._creditCardApprovalCode) _
            .AddChild(e.Status.BindToComboBox(), Function(v) v._status) _
            .AddChild(e.OnlineOrderFlag.BindToCheckBox(), Function(v) v._onlineOrderFlag) _
            .AddChild(e.Comment.BindToTextBox(), Function(v) v._comment)
        If Not isNew Then
            result.AddChild(e.SalesOrderNumber.BindToTextBlock(), Function(v) v._salesOrderNumber)
        End If
        Return result
    End Function

    <System.Runtime.CompilerServices.Extension>
    Public Function BindToSalesOrderFooterBox(Of T As {SalesOrderDetail, New})(e As SalesOrderBase(Of T)) As RowCompositeBinding(Of SalesOrderFooterBox)
        Return New RowCompositeBinding(Of SalesOrderFooterBox)() _
            .AddChild(e.SubTotal.BindToTextBlock("{0:C}"), Function(v) v._subTotal) _
            .AddChild(e.Freight.BindToTextBox(), Function(v) v._freight) _
            .AddChild(e.TaxAmt.BindToTextBox(), Function(v) v._taxAmt) _
            .AddChild(e.TotalDue.BindToTextBlock("{0:C}"), Function(v) v._totalDue)
    End Function

    <System.Runtime.CompilerServices.Extension>
    Public Function BindToAddressBox(e As Address) As RowCompositeBinding(Of AddressBox)
        Return New RowCompositeBinding(Of AddressBox)() _
            .AddChild(e.AddressLine1.BindToTextBlock(), Function(v) v._addressLine1) _
            .AddChild(e.AddressLine2.BindToTextBlock(), Function(v) v._addressLine2) _
            .AddChild(e.City.BindToTextBlock(), Function(v) v._city) _
            .AddChild(e.StateProvince.BindToTextBlock(), Function(v) v._stateProvince) _
            .AddChild(e.CountryRegion.BindToTextBlock(), Function(v) v._countryRegion) _
            .AddChild(e.PostalCode.BindToTextBlock(), Function(v) v._postalCode)
    End Function
End Module